board/app/models/user.php

```diff
@@ -103,37 +142,83 @@ public function login()
             'username' => $this->username,
             'password' => md5($this->password)
         );
+        $user = $db->row("SELECT id, username FROM user WHERE BINARY username = :username AND BINARY password = :password", $params);
 
-    $user = $db->row("SELECT id, username FROM user WHERE BINARY username = :username && BINARY password = :password", $params);
-
-        if(!$user) 
-        {
-            $this->is_validated = false; 
+        if(!$user) {
             throw new RecordNotFoundException('No Record Found');
         }
 
-    $_SESSION['user_id'] = $user['id'];
-    $_SESSION['username'] = $user['username'];
+        $_SESSION['user_id'] = $user['id'];
+        $_SESSION['username'] = $user['username'];
     }
 
-    public function is_password_match()
+    public function update()
     {
-        return $this->password == $this->confirm_password;
+        if (!$this->validate()) {
+            throw new ValidationException('Invalid user credentials');
+        }
+
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $db->update(
+                'user', array(
+                    'username' => $this->username,
+                    'first_name' => $this->first_name,
+                    'last_name' => $this->last_name,
+                    'password' => $this->password
+                    ),
+                array('id' =>$this->user_id)
+                );
+            $db->commit();
+        } catch(Exception $e) {
+            $db->rollback();
+            throw $e;
+        }
     }
 
+    public static function get($user_id)
+    {
+        $db = DB::conn();
+        $row = $db->row('SELECT * FROM user WHERE id = ?', array($user_id));
+        
+        return $row;
+    } 
 
-    public function is_username_exist()
+    public static function getOtherUsers($offset, $limit, $id)
     {
+        $users = array();
         $db = DB::conn();
-        $username_exist = $db->row("SELECT username FROM user WHERE username = ?", array($this->username));
-        return (!$username_exist);
+
+        if (!is_int($offset) || !is_int($limit)) { 
+            throw new NotIntegerException; 
+        }
+
+        $rows = $db->rows("SELECT * FROM user where id != ? LIMIT {$offset}, {$limit}", array($id));
+            
```

Comment:
Insert checking here if there are any data returned from the DB. Thrown an exception if it's empty.
It's the reason why we put:
```
if (!$row) {
    throw new RecordNotFoundException('no record found');
}
```